drivers/adc/ad463x/iio_ad463x.c

```diff
@@ -91,31 +151,194 @@ static int32_t _iio_ad463x_prepare_transfer(struct iio_ad463x *desc,
+ * @return ret    - Result of the reading procedure.
+ *		    In case of success, the size of the read data is returned.
+*/
+static int ad463x_iio_store_scale(void *dev, char *buf, uint32_t len,
+				  const struct iio_ch_info *channel, intptr_t priv)
+{
+	struct iio_ad463x *iio_desc = dev;
+	struct ad463x_dev *ad463x_dev;
+	int32_t gain_idx;
+	int ret;
+	int32_t vals[2];
+
+	ad463x_dev = iio_desc->ad463x_desc;
+	/** Scale only configurable if PGIA is available */
+	if(!ad463x_dev->has_pgia)
+		return -EINVAL;
+	iio_parse_value(buf, IIO_VAL_FRACTIONAL, &vals[0], &vals[1]);
+
+	ret = ad463x_calc_pgia_gain(vals[0], vals[1], ad463x_dev->vref,
+				    ad463x_dev->real_bits_precision, &gain_idx);
+	if (ret < 0)
+		return ret;
+	return ad463x_set_pgia_gain(ad463x_dev, gain_idx);
+}
+
+/**
+ * @brief Handles the read request for scale available attribute.
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ * @return ret    - Result of the reading procedure.
+ *		    In case of success, the size of the read data is returned.
+*/
+static int ad463x_iio_read_scale_avail(void *dev, char *buf,
+				       uint32_t len, const struct iio_ch_info *channel, intptr_t priv)
+{
+	int32_t vals[2];
+	struct iio_ad463x *iio_desc = dev;
+	struct ad463x_dev *ad463x_dev;
+	ad463x_dev = iio_desc->ad463x_desc;
+	char *buffer = buf;
+
+	if(ad463x_dev->has_pgia) {
+		/** Go through the values in the table and add them to the buffer */
+		for (uint8_t i = 0; i < NO_OS_ARRAY_SIZE(ad463x_dev->scale_table); i++) {
+			vals[0] = ad463x_dev->scale_table[i][0];
+			vals[1] = ad463x_dev->scale_table[i][1];
+			buffer += iio_format_value(buffer, len, IIO_VAL_INT_PLUS_NANO, 2, vals) + 1;
+			strcat(buf, " ");
+		}
+		buf[strlen(buf)] = '\0';
+		return strlen(buf);
+	}
+
+	vals[0] = (ad463x_dev->vref * 2) / 1000;
+	vals[1] = ad463x_dev->capture_data_width;
+	return iio_format_value(buf, len, IIO_VAL_FRACTIONAL_LOG2, 2, vals);
+}
+
+/**
+ * @brief Reads the number of given samples for the selected channels
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param samples - Number of samples to be returned
+ * @return ret    - Result of the reading procedure.
+ * 					In case of success, the size of the read data is returned.
+*/
 static int32_t _iio_ad463x_read_dev(struct iio_ad463x *desc, uint32_t *buff,
 				    uint32_t nb_samples)
 {
 	int ret;
-	uint32_t data[nb_samples], i, j, ch;
+	uint16_t total_samples = nb_samples * 2;
+	uint32_t *data;
+	uint32_t i, j, ch;
 
 	if (!desc)
 		return -EINVAL;
 
-	ret = ad463x_read_data(desc->ad463x_desc, data, nb_samples);
-	if(ret)
-		return ret;
+	data = (uint32_t *)no_os_calloc(1, sizeof(uint32_t) * total_samples * 2);
```

Comment:
the first parameter of `calloc()` should be the number of element, and the second one is the size of each element.
So, the proper way to write this would be:

`no_os_calloc(total_samples * 2, sizeof(*data));`

Reply:
noted, thank you.